### ISSUE TITLE
Set up signing and notarization

### ISSUE THREAD

**Issue by strogonoff** [paneron/paneron#2]
macOS build [reports](https://github.com/paneron/paneron/runs/1199329363?check_suite_focus=true):

> Either APPLE_ID and APPLE_ID_PASSWORD, or API_KEY_ID and API_KEY_ISSUER_ID must be defined

**Comment by strogonoff** [comment]
On macOS, complains that developer is unidentified and suggests deleting the file. Requires right-click -> Open.

**Comment by ronaldtse** [comment]
All secrets are now properly set. Can you try again?

**Comment by strogonoff** [comment]
@ronaldtse Thank you!

It seems that one more action is required for macOS build:

```
1 package(s) were not uploaded because they had problems:
/…/org.paneron.desktop.itmsp - Error Messages: You must first sign the relevant contracts online. (1048)
2020-10-12 23:56:27.234 altool[1300:11103] *** Error: Unable to notarize app.
2020-10-12 23:56:27.234 altool[1300:11103] *** Error: code 1048 (You must first sign the relevant contracts online. (1048))
```

**Comment by ronaldtse** [comment]
@strogonoff Done!

**Comment by ronaldtse** [comment]
Please close when confirmed, thanks!

**Comment by strogonoff** [comment]
@ronaldtse No more contract signing error, but there seem to be a notarization problem and exact issue should be available at Apple—please check out [this part of our build log](https://github.com/paneron/paneron/runs/1250549619?check_suite_focus=true#step:5:663)?

**Comment by strogonoff** [comment]
@ronaldtse I sent you a link privately, if there’s any useful troubleshooting information there please let me know

**Comment by strogonoff** [comment]
Confirming it works on macOS.

Windows test pending.

There is still a problem with Ubuntu. It is possible that the lack of some sort of verification of the Snap package makes the app “untrusted”, and that’s why repository passwords cannot be saved and retrieved by the app. (This has been an issue with Glossarist Desktop, too.)

```
Remote error from secret service: org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.444" (uid=1000 pid=2806 comm="/snap/paneron/x1/paneron.bin --no-sandbox --no-san" label="snap.paneron.paneron (enforce)") interface="org.freedesktop.Secret.Service" member="OpenSession" error name="(unset)" requested_reply="0" destination=":1.20" (uid=1000 pid=1607 comm="/usr/bin/gnome-keyring-daemon --daemonize --login " label="unconfined")
```

**Comment by ronaldtse** [comment]
I haven’t made the snap account. That’s probably why. Will do so today.

**Comment by strogonoff** [comment]
> I haven’t made the snap account. That’s probably why. Will do so today.

@ronaldtse I found this comment: https://github.com/bitwarden/desktop/issues/215#issuecomment-623436919

It suggests a user action may work. It recommends:

1. Go to Ubuntu Software
2. Search for the app throwing the error
3. Click Permissions
4. Turn ON Read, add, change, or remove saved passwords

I will check if that works, and if it does then we can simply give those instructions to Ubuntu users.

**Comment by strogonoff** [comment]
For Ubuntu: fixed in 2170368ae83fa14e087ed703e41a9b044a183e4b and with following extra installation instruction for Ubuntu users:

> Note for Ubuntu users:

> After installing Paneron from snap, run the following command:

> ```
> snap connect paneron:password-manager-service
> ```
